### ISSUE TITLE
Dots aren't happening on first click

### ISSUE THREAD

**Issue by rogerhutchings** [zooniverse/shakespeares_world#314]
When you load up a subject, click text and start transcribing, the first two clicks don't produce dots, although it does open the transcription window. Subsequent pairs of clicks produce dots as normal.

**Comment by eatyourgreens** [comment]
In Firefox, I'm finding that I have to click five or six times before the transcription window appears.

**Comment by shaunanoordin** [comment]
## Investigation Notes

When a Classification exists but but the Blue Dots are invisible, it looks like this:
```
<g class="text-annotation ng-isolate-scope" ng-class="{ '-incomplete': !data.complete }" text-annotation="" data="annotation" ng-repeat="annotation in vm.annotations | filter:{type:'text'}:true" style="touch-action: pan-y; user-select: none; -webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);">

    <circle draggable="data.startPoint" class="text ng-isolate-scope" ng-show="data.startPoint" ng-attr-cx="{{ data.startPoint.x }}" ng-attr-cy="{{ data.startPoint.y }}" ng-attr-r="{{ ::r }}" cx="247.92" cy="588.99" r="5"></circle>

    <circle draggable="data.endPoint" ng-show="data.endPoint" class="text ng-isolate-scope" ng-attr-cx="{{ data.endPoint.x }}" ng-attr-cy="{{ data.endPoint.y }}" ng-attr-r="{{ ::r }}" cx="1057.32" cy="583.41" r="5"></circle>

</g>
```

Conversely, when a Classification exists but the Blue Dots are visible, it looks like this...
```
<g class="text-annotation" ng-class="{ '-incomplete': !data.complete }" text-annotation="" data="annotation" ng-repeat="annotation in vm.annotations | filter:{type:'text'}:true" style="touch-action: pan-y; user-select: none; -webkit-user-drag: none; -webkit-tap-highlight-color: rgba(0, 0, 0, 0);">

    <circle draggable="data.startPoint" class="text" ng-show="data.startPoint" ng-attr-cx="{{ data.startPoint.x }}" ng-attr-cy="{{ data.startPoint.y }}" ng-attr-r="{{ ::r }}" cx="421.31" cy="1122.46" r="5"></circle>

    <circle draggable="data.endPoint" ng-show="data.endPoint" class="text" ng-attr-cx="{{ data.endPoint.x }}" ng-attr-cy="{{ data.endPoint.y }}" ng-attr-r="{{ ::r }}" r="5" cx="697.71" cy="1118.24"></circle>

</g>
```

Note the presence/absence of `ng-isolate-scope`. Query: what's does `ng-isolate-scope` mean?

Further observations: a Classification is extant but invisible if:
* It is the first Classification of the Subject, and/or
* It was loaded from storage (i.e. "remembered" from a past session)

**Comment by shaunanoordin** [comment]
## Investigation Notes
@simoneduca and me have discovered what's causing the problem, but no explanation of _why_.

In (app/modules/transcribe/annotations/text.directive.js)[https://github.com/zooniverse/shakespeares_world/blob/master/app/modules/transcribe/annotations/text.directive.js#L13] the line `replace: true` causes the Text Annotations to not render properly.

Changing this to `replace: false` (the default setting) will fix this on MOST browsers. See Testing Notes for observations.

## General Notes
* Related to #290 
* Related to #298 ?
* For reference, the `replace` feature in AngularJS has been deprecated; basically, `replace: true` has been known to be pretty scrappy.

## Testing Notes
Question: "Does changing `replace: true` to `replace: false` allow the Shakespeare's World app to work as intended?"

Tests:
* **Add** Text Annotations - Create a new Text Annotation on the Subject. Click on a start point (a pulsing red dot should appear) then an end point (the dots should turn blue). A Transcription Window should open. Type in some text then click 'Save/Done'. The new Text Annotation should appear as a pair of blue dots upon completion.
* **Remember** Text Annotations - Reload page. The previous annotation(s) should be visible as blue dots.
* **Edit** Text Annotations - Click on an existing Text Annotation (blue dot). Click Modify on the menu that appears. The Transcription Window should open, allowing editing of the original transcription. Click 'Save/Done', the window should close.
* **Delete** Text Annotations - Click on an existing Text Annotation (blue dot). Click Modify on the menu that appears. The Text Annotation (pair of blue dots) should disappear.

Tests done on localhost:3000/?env=production

Results:
* OSX + Chrome 54: OK
* OSX + FF 49: OK
* OSX + Safari 9: OK
* Win7 + IE11: ERROR - NO dots appear whatsoever
  * If we revert to `replace: true`, the original problems arise, so we can't ignore it.

Since the AngularJS code is similar to AnnoTate, I've run some tests on AnnoTate to confirm whether `replace: false` will also fix the same blue dot problems on that project.

Results (Annotate ver)
* OSX + Chrome 54: OK
* OSX + FF 49: OK
* OSX + Safari 9: OK
* Win7 + IE11: ERROR - The project won't load whatsoever; it seems that Promise is undefined.
  * Same on localhost as anno.tate.org.uk

**Comment by simoneduca** [comment]
Given the frustration that this has been and still is causing among our volunteers, I think we should push the fix out and worry about IE. In my opinion, it's better to tell them to use a different browser than giving them something that doesn't quite work. So I'm going ahead.

**Comment by rogerhutchings** [comment]
IE - is it not just a matter of adding a polyfill to begin with?

**Comment by simoneduca** [comment]
Might be; but we can investigate that once this is fixed, right?

**Comment by simoneduca** [comment]
@rogerhutchings btw iirc the polyfill was added at some point... don't know what happened to it. I assume you mean the promises polyfill?

**Comment by rogerhutchings** [comment]
yup

**Comment by shaunanoordin** [comment]
The issue with IE11 needs to be broken down to two parts:

1. **Shakespeare's World** (and probably Annotate): we'll have issues rendering the blue dots _whether or not_ we have the `replace: true/false` fix, so this require a deeper dig into the code.

2. **Annotate** only: the whole project doesn't have a Promise polyfill. It may have been the reason behind this PR: https://github.com/zooniverse/AnnoTate/pull/213

In any case, I agree with @simoneduca: merge the fixes #316 and #317 first so FF/Chrome/Safari users are OK, then fix the problem with IE.
